src/sentry/search/snuba/executors.py

```diff
@@ -1692,9 +1703,10 @@ def query(
                 Condition(Column("timestamp", joined_entity), Op.GTE, start),
                 Condition(Column("timestamp", joined_entity), Op.LT, end),
             ]
+
             having = []
             # if we need to prefetch from postgres, we add filter by the group ids
-            if group_ids_to_pass_to_snuba is not None:
+            if group_ids_to_pass_to_snuba is not None and not too_many_candidates:
```

Comment:
going back to the comment above, if you reset group_ids_to_pass_to_snuba to None when too_many_candidates, not need to change anything here